### PR TITLE
remove active waiting in party

### PR DESCRIPTION
**Pull request by kostekIV**

### CODE REVIEW

File: finality-aleph/src/party/mod.rs [hunk 1]
```diff
@@ -521,7 +527,8 @@ where
                         }
                         start_next_session_network = None;
                     } else {
-                        start_next_session_network = Some(Delay::new(NEXT_SESSION_NETWORK_START_ATTEMPT_PERIOD));
+                        // should not happen
```
Comment (timorl) [inline finality-aleph/src/party/mod.rs:530]: Then log a warning.

File: finality-aleph/src/party/mod.rs [hunk 2]
```diff
@@ -478,14 +481,17 @@ where
                 },
                 Some(_) = async {
                     match &mut start_next_session_network {
-                        Some(start_next_session_network) => {
-                            start_next_session_network.await;
+                        Some(notification) => {
+                            if let Err(e) = notification.await {
+                                error!(target: "aleph-party", "Error with subscription {:?}", e);
```
Comment (timorl) [inline finality-aleph/src/party/mod.rs:486]: This should only be a `warn`, something went wrong, but we can go on.

File: finality-aleph/src/party/mod.rs [hunk 3]
```diff
@@ -478,14 +481,17 @@ where
                 },
                 Some(_) = async {
                     match &mut start_next_session_network {
-                        Some(start_next_session_network) => {
-                            start_next_session_network.await;
+                        Some(notification) => {
+                            if let Err(e) = notification.await {
```
Comment (timorl) [inline finality-aleph/src/party/mod.rs:485]: Perhaps return `next_session_authorities` here already.

**Review by timorl — CHANGES_REQUESTED** [review]
I really like this improvement, nice.

**Review by Marcin-Radecki — APPROVED** [review]
I have nothing more to add other than @timorl's comments. Nice rework.

File: finality-aleph/src/party/mod.rs (hunk 1 above)
Comment (kostekIV) [reply finality-aleph/src/party/mod.rs]: else is no longer there

File: finality-aleph/src/party/mod.rs (hunk 2 above)
Comment (kostekIV) [reply finality-aleph/src/party/mod.rs]: done

File: finality-aleph/src/party/mod.rs (hunk 3 above)
Comment (kostekIV) [reply finality-aleph/src/party/mod.rs]: done

File: finality-aleph/src/party/mod.rs [hunk 4]
```diff
@@ -39,12 +39,15 @@ use sp_runtime::traits::{Block, Header};
 use std::{collections::HashSet, default::Default, marker::PhantomData, sync::Arc, time::Duration};
 
 mod task;
+
 use task::{Handle, Task};
+
```
Comment (timorl) [inline finality-aleph/src/party/mod.rs:44]: In other modules we put all these `use`s below a block of `mod`s, the sudden empty lines here are kinda ugly, so I would prefer if you did that.

The empty line below, separating the `mod`s from the `use`s is a fine addition ofc.

**Review by timorl — APPROVED** [review]
Nice, tiny whitespace nit, but other than that LGTM.

File: finality-aleph/src/party/mod.rs (hunk 4 above)
Comment (kostekIV) [reply finality-aleph/src/party/mod.rs]: I think fixed, could you check if this is what you meant?

**Pull request merged**